### ISSUE TITLE
Permission query algorithm should be able to fail

### ISSUE THREAD

**Issue by jyasskin** [w3c/permissions#135]
The [permission query algorithm](https://w3c.github.io/permissions/#permission-query-algorithm) doesn't currently accommodate features that need to validate their descriptors, which can cause the query to fail. Like with the [request algorithm](https://w3c.github.io/permissions/#permission-request-algorithm), we should allow the customized query algorithm to throw.

**Comment by marcoscaceres** [comment]
Throwing is kinda ugly tho - and such checks shouldn't be classified as exceptional - can we come up with a more graceful way to do the check?  

Maybe related to https://github.com/w3c/permissions/issues/136 ?

**Comment by jyasskin** [comment]
I think passing an invalid descriptor to `query()` *is* exceptional, in the same way that passing `3` would be. Passing an un-implemented `PermissionName` might not be exceptional, but that's not what I'm requesting here.

**Comment by miketaylr** [comment]
This does seem useful, in theory. But is it true that no feature has requested it? Maybe we wait until then to spec it out?

Or, do we know of examples/use cases today that are working around, or held back by the lack of this ability?

**Comment by marcoscaceres** [comment]
Agree. We discussed that we can amend "A permission query algorithm" to incorporate this if/when needed by a particular descriptor. Right now these are checked by WebIDL. 

We might need to do a thorough check of various permissions to make sure other specs are not actually doing these kinds of checks.
